### ISSUE TITLE
Can't parse dynamic imports

### ISSUE THREAD

**Issue by externuz** [i18next/i18next-parser#155]
- Is this a bug or a feature request?
`bug`
- What version are you running (`0.x` or `1.x`)? 
`"i18next-parser": "1.0.0-beta34"`
`"acorn-jsx": "4.x"`
`"acorn-stage3": "0.6.0"`

- Which runner (cli, gulp...)?
`cli`

## What I'm trying to do?
I am trying to parse my `ts` and `tsx` files

## How am I doing it?
`i18next \"./src/**/*.{ts,tsx}\" -o \"./locales\"`

Here is my config file:

```
module.exports = {
  createOldCatalogs: true,
  keepRemoved: true,
  locales: ["en", "ru"],
  verbose: true,
};
```

The command I'm running:

```
i18next \"./src/**/*.{ts,tsx}\" -o \"./locales\"
```

## What was the result?

```
  [read]  /Users/externuz/code/smpio/smmplanner-front/src/api/smp/apiSmpSocialProvidersGet.ts
/Users/externuz/code/smpio/smmplanner-front/node_modules/i18next-parser/dist/parser.js:70
        }} catch (err) {_didIteratorError = true;_iteratorError = err;} finally {try {if (!_iteratorNormalCompletion && _iterator.return) {_iterator.return();}} finally {if (_didIteratorError) {throw _iteratorError;}}}
                                                                                                                                                                                                  ^

SyntaxError: Unexpected token (5:38)
    at Parser.pp$4.raise (/Users/externuz/code/smpio/smmplanner-front/node_modules/acorn/dist/acorn.js:2757:13)
    at Parser.pp.unexpected (/Users/externuz/code/smpio/smmplanner-front/node_modules/acorn/dist/acorn.js:647:8)
    at Parser.pp$3.parseExprAtom (/Users/externuz/code/smpio/smmplanner-front/node_modules/acorn/dist/acorn.js:2196:10)
    at Parser.parseExprAtom (/Users/externuz/code/smpio/smmplanner-front/node_modules/acorn-jsx/inject.js:399:24)
    at Parser.pp$3.parseExprSubscripts (/Users/externuz/code/smpio/smmplanner-front/node_modules/acorn/dist/acorn.js:2047:19)
    at Parser.pp$3.parseMaybeUnary (/Users/externuz/code/smpio/smmplanner-front/node_modules/acorn/dist/acorn.js:2024:17)
    at Parser.pp$3.parseAwait (/Users/externuz/code/smpio/smmplanner-front/node_modules/acorn/dist/acorn.js:2742:24)
    at Parser.pp$3.parseMaybeUnary (/Users/externuz/code/smpio/smmplanner-front/node_modules/acorn/dist/acorn.js:2008:17)
    at Parser.pp$3.parseExprOps (/Users/externuz/code/smpio/smmplanner-front/node_modules/acorn/dist/acorn.js:1966:19)
    at Parser.pp$3.parseMaybeConditional (/Users/externuz/code/smpio/smmplanner-front/node_modules/acorn/dist/acorn.js:1949:19)
error Command failed with exit code 1.
```

## What was I expecting?

I expected successful file parsing but got this error

**Comment by externuz** [comment]
I tried to use this config, but no changes

```
const injectAcornStaticClassPropertyInitializer = require("acorn-static-class-property-initializer/inject");
const injectAcornStage3 = require("acorn-stage3/inject");
const injectAcornEs7 = require("acorn-es7");

module.exports = {
  createOldCatalogs: true,
  keepRemoved: true,
  lexers: {
    tsx: [
      {
        acorn: {
          injectors: [
            injectAcornStaticClassPropertyInitializer,
            injectAcornStage3,
            injectAcornEs7,
          ],
          plugins: {
            staticClassPropertyInitializer: true,
            stage3: true,
            es7: true,
          },
        },
      },
    ],
  },
  locales: ["en", "ru"],
  sort: true,
  verbose: true,
};
```

**Comment by tquiroga** [comment]
Same problem for me, I'm looking into `acorn-dynamic-import` right now

**Comment by bitttttten** [comment]
It's strange as `acorn-stage3` should be including dynamic imports already:

https://github.com/acornjs/acorn-stage3/blob/f88a220fa6fbf20e77fdc0f5cb100e44f4cd2d99/index.js#L5

But where does 'acorn-stage3/inject' come from? The package does not have this file, only an index.js file.

**Comment by bitttttten** [comment]
I figured it out. You must install `acorn-stage3@0.6.0`.

So I ran: `yarn add acorn-stage3@0.6.0` then:

```js
const gulp = require('gulp')
const i18nextParser = require('i18next-parser').gulp
const stage3Injector = require('acorn-stage3/inject')

gulp.task('i18next', function() {
    gulp.src('./**/*.tsx')
        .pipe(
            new i18nextParser({
                locales: ['en', 'nl'],
                output: './$LOCALE/$NAMESPACE.json',
                verbose: true,
                lexers: {
                    tsx: [
                        {
                            lexer: 'TypescriptLexer',
                            attr: 'i18nKey',
                            tsOptions: {
                                jsx: 'Preserve',
                                target: 'esnext',
                            },
                            acorn: {
                                injectors: [stage3Injector],
                                plugins: { stage3: true },
                            },
                        },
                    ],
                },
            }),
        )
        .pipe(gulp.dest('/'))
})
```

**Comment by connor4312** [comment]
It might be worth investigating using the TypeScript compiler/ast for all code. It's a superset of Js/Jsx, and it supports modern constructs as soon as they pick up adoption. It'd also let us streamline the lexers a bit, using one toolchain instead of many + manual patching in acorn.

**Comment by bitttttten** [comment]
It would seem to simply the lexers a lot! And way less to maintain. Sharing configs would become a lot easier too.

**Comment by connor4312** [comment]
@karellm thoughts? I'm happy to put in a PR if you've interest. (We're still using this library at my company :) )

**Comment by karellm** [comment]
Sorry guys, I've not been very present recently, I don't really code anymore but if you write a PR to improve this mess, I will absolutely review it!

Long story short, last time I checked and touched this project, I couldn't update to the latest acorn because many plugins were lagging behind. This probably changed.

**Comment by uqee** [comment]
sadly, has happened to me also ) kinda solved it by parsing `*.tsx` files only, where I don't have dynamic imports and - hopefully - I don't have translations in `*.ts` files

**Comment by karellm** [comment]
I'm very aware of the situation. Here is what's happening:

This project uses `acorn@5.x` and as a result we are stuck with old plugins: `acorn-stage3@0.6.0`, `acorn-es7@0.1.0` and `acorn-jsx@4.1.1`. Don't try to update them, it will fail.

**Why don't I update?** Because the new version of the plugins are not supporting `acorn-walk`. So it's a pita to update this parser.

**What do we do?** We patiently wait for the "acorn" situation to improve or one of you invest some time to make a PR with a different parser than acorn.

**Comment by karellm** [comment]
#180 was merged and `1.0.0-beta36` published. Acorn is dead, long live Typescript. Thanks a lot @vincaslt
